cobalt/browser/browser_module.cc

```diff
@@ -539,6 +539,23 @@ void BrowserModule::Navigate(const GURL& url_reference) {
   // https://fetch.spec.whatwg.org/commit-snapshots/8f8ab504da6ca9681db5c7f8aa3d1f4b6bf8840c/#http-fetch
   bool can_start_service_worker = url.SchemeIsHTTPOrHTTPS();
   auto service_worker_started_event = std::make_unique<base::WaitableEvent>();
+  watchdog::Watchdog* watchdog = watchdog::Watchdog::GetInstance();
```

Comment:
Just an observation, no action needed. This "Navigate" function is getting really long, and it does a lot of unrelated things in sequence.

I would recommend considering breaking it out into multiple smaller functions in future so that the code itself here becomes more readable.

e.g 
```
void BrowserModule::Navigate(const GURL& url_reference) {
    NavigateResolveUrl();
    NavigateResetWebModules();
    NavigateSetupWorkers();
    .... etc
```
Or some other refactor that makes it a bit more obvious as to what all happens in this sequence.

Reply:
See: https://github.com/youtube/cobalt/pull/1336